### PR TITLE
[docs][auth session] Added Auth0 support

### PR DESCRIPTION
**Pull request by shehbajdhillon**
# Why

Auth0 is a popular authorization as a service especially amongst solo developers and small teams. However, Auth0 lacks documentation on how to use its service with Expo's AuthSession module. The purpose of this PR is to add documentation on how to use Auth0 as an auth provider with AuthSession using authorization code flow, implicit flow, and PKCE flow. The reason why I have added a PKCE flow separately is that it is extremely simple to obtain a token pair using PKCE flow compared to traditional server-side authorization code flow.

I spent a week trying to get Auth0 to work with AuthSession due lack of documentation and tutorials available on the internet. This PR will definitely save a lot of time for future devs who intend to use Auth0 with AuthSession,

# How

Adding documentation on how to use Auth0 as a provider in AuthSession.

# Test Plan

I ran prettier and lint on the files I edited and tested the changes I made by running the dev server locally.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/guides/authentication.md [hunk 1]
```diff
@@ -109,6 +110,233 @@ export default function App() {
...
+        audience: 'AUTH0_AUDIENCE',
+      },
+    },
+    discovery,
+  );
+
+  useEffect(() => {
+    if (response?.type === 'success') {
+      //The id_token is used as the access token in implicit flow.
+      const { id_token } = response.params;
```
Comment (EvanBacon) [inline docs/pages/guides/authentication.md:230]: Generally we use the magic comments here (they glow and have pop-over):

```
/* @info The <code>id_token</code> is used as the access token in implicit flow. */
const { id_token } = response.params;
/* @end */
```

**Review by EvanBacon — COMMENTED** [review]
Could you resize the image to 256x256, and try to get the size around 4-8kb like the others?

File: docs/pages/guides/authentication.md [hunk 2]
```diff
@@ -109,6 +110,233 @@ export default function App() {
...
+WebBrowser.maybeCompleteAuthSession();
+
+const discovery = {
+  authorizationEndpoint: 'https://<AUTH0_DOMAIN>/authorize',
+  tokenEndpoint: 'https://<AUTH0_DOMAIN>/oauth/token',
+};
+
+const App: React.FC = () => {
+  const useProxy = Platform.select({ web: false, default: true });
+  const [randomValue] = useState(Math.random().toString(36).substring(2, 15));
```
Comment (EvanBacon) [inline docs/pages/guides/authentication.md:145]: This is insecure, maybe try using `import { generateHexStringAsync } from 'expo-auth-session/build/PKCE'`, could be a good PR to expose that as a hook in the API.

File: docs/pages/guides/authentication.md [hunk 3]
```diff
@@ -109,6 +110,233 @@ export default function App() {
...
```
Comment (EvanBacon) [inline docs/pages/guides/authentication.md:210]: same as above, also maybe want to `useMemo` for things like this in the future.

**Comment by soulfresh** [comment]
@dshehbaj Does your solution work on web? You example works great for me on iOS but the authentication popup never closes on web. Instead, the popup redirects back to my app and I end up with my login page showing in both my browser tab and the popup window. I'm wondering if you've encountered this or if I'm missing something on my end.

**Comment by soulfresh** [comment]
Nevermind, it looks like I was placing the `WebBrowser.maybeCompleteAuthSession()` in the wrong place. Thanks for your PR. That really helped!

**Comment by ogmzh** [comment]
Almost 2 years from this PR, is it ever getting merged/updated?

**Comment by amandeepmittal** [comment]
Since there's been no activity on this PR, I'm closing it since the contents of the PR would need to be revise to make sure the example works and also from doc's perspective, it's follows our writing style guide.

**Pull request closed without merging**